### PR TITLE
Load mini cart scripts by default on touch-enabled (mobile) devices

### LINKED ISSUE #48153: Minicart drawer close button missing on mobile pages
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

On mobile, when accessing product pages, if I try to open minicart, the close button is invisible.

Clicking on some focusable element will show it.

This is happening only on mobile and only at page load.

### Expected behavior

Close button should be always visible

### Actual behavior

Close button not visible on page load

### Steps to reproduce

1. On mobile go to a product page
2. On page load open minicart.

### WordPress Environment

<details><summary>System Status Report</summary>

```
### WordPress Environment ###

WordPress address (URL): http://localhost:10005
Site address (URL): http://localhost:10005
WC Version: 9.0.0-rc.1
Legacy REST API Package Version: The Legacy REST API plugin is not installed on this site.
Action Scheduler Version: ✔ 3.7.4
Log Directory Writable: ✔
WP Version: 6.5.3
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: nginx/1.16.0
PHP Version: 7.4.3
PHP Post Max Size: 1,000 MB
PHP Time Limit: 1200
PHP Max Input Vars: 4000
cURL Version: 7.68.0
OpenSSL/1.1.1d

SUHOSIN Installed: –
MySQL Version: 8.0.16
Max Upload Size: 300 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 9.0.0
WC Database Prefix: wp_
Total Database Size: 4.25MB
Database Data Size: 2.54MB
Database Index Size: 1.71MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 1.45MB + Index: 0.06MB + Engine InnoDB
wp_postmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_posts: Data: 0.06MB + Index: 0.06MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.05MB + Index: 0.02MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.06MB + Index: 0.00MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_addresses: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_operational_data: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_orders: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_wc_orders_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 1
page: 7
post: 2
product: 36
product_variation: 18
revision: 5
shop_order_placehold: 1
wp_global_styles: 1
wp_navigation: 1
wp_template: 2

### Security ###

Secure connection (HTTPS): ❌
					Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ❌Error messages should not be shown to visitors.

### Active Plugins (3) ###

Archimedia Blocks: by Francesco Pepe – 0.1.0
WooCommerce: by Automattic – 9.0.0-rc.1
WP Reset: by WebFactory Ltd – 2.02

### Inactive Plugins (75) ###

AI Power: Complete AI Pack - Powered by GPT-4: by Senol Sahin – 1.8.65
BetterLinks: by WPDeveloper – 2.0.0
Block Visibility: by Nick Diego – 3.4.0
Brave Conversion Engine: by Brave – 0.6.8
Breakdance: by Breakdance – 1.7.0
bunny.net: by bunny.net – 2.2.7
CasertaKeste: by Francesco – 1.0.0
ClassifAI: by 10up – 2.5.1
ClickWhale: by ClickWhale – 2.1.3
ConvertKit: by ConvertKit – 2.4.9
Create Block Theme: by WordPress.org – 2.1.4
CRM Perks HelpDesk Integration: by CRM Perks – 1.1.4
Database for WPforms: by wpdebuglog – 1.0.4
Digital License Manager: by CodeVerve – 1.6.3
FG PrestaShop to WooCommerce Premium: by Frédéric GILLES – 4.37.1
FiboSearch - AJAX Search for WooCommerce: by FiboSearch Team – 1.28.0
Fluent Forms: by Contact Form - WPManageNinja LLC – 5.1.18
Folders: by Premio – 3.0.3
Formello: by Formello – 2.5.1
Formello Docs: by Tropicalista – 0.1.0
Formello Pro: by Formello – 1.0.4
Formello Templates: by Formello – 1.0.0
FormFacade: by FormFacade – 1.2.9
Formidable Forms: by Strategy11 Form Builder Team – 6.9.1
Freshdesk Official: by Freshworks – 2.4.0
GenerateBlocks: by Tom Usborne – 1.9.1
GenerateBlocks Pro: by Tom Usborne – 1.6.0
GetPaid: by AyeCode Ltd – 2.8.7
Google Analytics Dashboard for WP (GADWP): by ExactMetrics – 7.27.0
GP Premium: by Tom Usborne – 2.1.2
Gravity Forms: by Gravity Forms – 2.6.4
Gutenberg: by Gutenberg Team – 18.4.1
HubSpot All-In-One Marketing - Forms, Popups, Live Chat: by HubSpot – 11.1.13
Kadence Conversions - Popups, slide-ins: by Kadence WP – 1.0.3
Kint PHP Debugger: by hellofromTonya – 2.0.2
Lemon Squeezy: by Lemon Squeezy – 1.3.0
Lemon Squeezy License Validator example: by Tropicalista – 0.1.0
MailPoet: by MailPoet – 4.51.0
Members: by MemberPress – 3.2.9
Mortgage: by Calcolo Rata Mutuo – 0.3.1
Ollie Pro: by buildwithollie – 1.0
Order Export & Order Import for WooCommerce: by WebToffee – 2.5.2
Otter – Page Builder Blocks & Extensions for Gutenberg: by ThemeIsle – 2.6.12
Pdf Embed: by Tropicalista – 0.4.3
Perfect Brands WooCommerce: by QuadLayers – 3.3.9
Popper: by Tropicalista – 0.7.2
Popper Pro: by Tropicalista – 0.1.0
Presto Player: by Presto Made
Inc – 2.3.3

Quill Forms: by quillforms.com – 3.6.2
Quotes Addon for GetPaid: by AyeCode Ltd – 2.0.6
Rank Math SEO with AI Best SEO Tools: by Rank Math – 1.0.219
Search Console: by Tropicalista – 3.0.3
Search Console JSON export addon: by Tropicalista – 0.1.0
Search Engine Insights: by Alin Marcu – 2.6.3
SimpleTOC - Table of Contents Block: by Marc Tönsing – 6.4.4
SliceWP: by SliceWP – 1.1.14
Software License Manager: by Tips and Tricks HQ – 4.5.6
SureCart: by SureCart – 2.27.1
SureTriggers: by SureTriggers – 1.0.47
The Icon Block: by Nick Diego – 1.8.0
Thrive Automator: by <a href="https://thrivethemes.com">Thrive Themes</a> – 1.20
Tripetto: by Tripetto – 8.0.1
Tropical Juice: by Tropicalista – 1.0.0
User Switching: by John Blackbourn & contributors – 1.7.3
UsersWP: by AyeCode Ltd – 1.2.10
UsersWP - Social Login: by AyeCode Ltd – 1.3.25
Variation Swatches for WooCommerce: by CartFlows – 1.0.8
WooCommerce Beta Tester: by WooCommerce – 2.3.0
WooCommerce Cart Abandonment Recovery: by CartFlows Inc – 1.2.27
WordPress Beta Tester: by WordPress Upgrade/Install Team – 3.5.5
WordPress Importer: by wordpressdotorg – 0.8.2
WPCode Lite: by WPCode – 2.1.12
WP Crontrol: by John Blackbourn – 1.16.3
WP Fusion Lite: by Very Good Plugins – 3.43.0
WP Help: by Mark Jaquith – 1.7.5

### Settings ###

API Enabled: –
Force SSL: –
Currency: EUR (€)
Currency Position: right_space
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔
HPOS feature enabled: ✔
Order datastore: Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore
HPOS data sync enabled: –

### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 days
Level threshold: –
Log directory size: 93 KB

### WC Pages ###

Shop base: #6 - /shop/
Cart: #7 - /cart/
Checkout: #8 - /checkout/
My account: #9 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Twenty-Four
Version: 1.1
Author URL: https://wordpress.org
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ❌ Not declared

### Templates ###

Overrides: –

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customize-store
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
product-custom-fields
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page
launch-your-store

Disabled Features: minified-js
new-product-management-experience
pattern-toolkit-full-composability
product-pre-publish-modal
settings
async-product-editor-category-field

Daily Cron: ✔ Next scheduled: 2024-06-04 16:00:55 +00:00
Options: ✔
Notes: 60
Onboarding: skipped

### Action Scheduler ###

Complete: 105
Oldest: 2024-06-03 16:01:58 +0000
Newest: 2024-06-04 15:16:29 +0000

Pending: 2
Oldest: 2024-06-04 15:19:22 +0000
Newest: 2024-06-04 16:01:58 +0000

Failed: 1
Oldest: 2024-06-03 22:11:48 +0000
Newest: 2024-06-03 22:11:48 +0000


### Status report information ###

Generated at: 2024-06-04 15:18:32 +00:00
```
</details>

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by opr**
### Changes proposed in this Pull Request:

This PR will cause the the dependencies for the mini cart to load immediately when the mini cart block loads on a device with touch enabled. Currently these dependencies only load when mouseovering or focusing the mini cart button, which is not a common use case on a touch device (it would require the user to hold their finger on the mini cart button)

It also adds a TS definition for `wcBlocksMiniCartFrontendDependencies` to prevent a TS warning in the same file.

Closes #48153

### How to test the changes in this Pull Request:

🌟  Perform these steps on mobile **and** desktop devices. Desktop is for regression catching, mobile is for checking the bug is fixed. 🌟 

1. Start with an empty cart.
2. Load a page with the mini-cart.
3. Open the mini cart and ensure it loads correctly with a useable close button.
4. Close the mini cart.
5. Repeat with an item in the cart.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/assets/js/blocks/mini-cart/frontend.ts [hunk 1]
```diff
@@ -158,6 +161,16 @@ window.addEventListener( 'load', () => {
...
+		if (
+			'ontouchstart' in window ||
+			navigator.maxTouchPoints > 0 ||
+			window.matchMedia( '(pointer:coarse)' ).matches
+		) {
+			loadScripts();
+		}
+
 		miniCartButton.addEventListener( 'mouseover', loadScripts );
 		miniCartButton.addEventListener( 'focus', loadScripts );
```
Comment (wavvves) [inline plugins/woocommerce-blocks/assets/js/blocks/mini-cart/frontend.ts]: Should we prevent adding listeners if it is touch enabled? I know the method bails early, still we don't need them triggering

Comment (opr) [reply plugins/woocommerce-blocks/assets/js/blocks/mini-cart/frontend.ts]: Good idea, not sure how I missed that.

**Review by wavvves — APPROVED** [review]
LGTM

**Comment by senadir** [comment]
Nice fix, but why not change the event to something compatible with mobile like touchdown?

**Pull request merged**